index 6986f77cc..5b072e1f3 100644
--- a/arch/arm64/boot/Makefile
+++ b/arch/arm64/boot/Makefile
@@ -33,6 +33,9 @@ ifdef RTIC_MPGEN
 DTB_OBJS += rtic_mp.dtb
 endif

+quiet_cmd_kpmpatch = PATCH   $@
+      cmd_kpmpatch = (cd $(obj) && $(srctree)/scripts/patch_kpm > /dev/null) && mv -f $(obj)/oImage $@
+
 rtic_mp.dtb: vmlinux FORCE
        $(RTIC_MPGEN) --objcopy="${OBJCOPY}" --objdump="${OBJDUMP}" \
        --binpath="" --vmlinux="vmlinux" --config=${KCONFIG_CONFIG} \
@@ -41,6 +44,9 @@ rtic_mp.dtb: vmlinux FORCE

 $(obj)/Image: vmlinux FORCE
        $(call if_changed,objcopy)
+ifeq ($(CONFIG_KPM),y)
+       $(call cmd,kpmpatch)
+endif

 $(obj)/Image.bz2: $(obj)/Image FORCE
        $(call if_changed,bzip2)